src/main/core/support/simulation_time.rs

```diff
@@ -0,0 +1,35 @@
+/*!
```

Comment:
Doesn't need to be this PR, but I think we'll eventually want some `From` implementations for converting between `SimulationTime` and `std::time::Instant` (and/or `std::time::Duration`)

Reply:
We aren't able to implement `From` since `SimulationTime` (a `u64`) and `std::time::Duration` are both types implemented outside of the current crate. I added a function for this conversion though (`fn sim_time_to_duration(time: c::SimulationTime) -> std::time::Duration`).